### DISCUSSION TITLE
How do I tell sqlc  to omit a field in the json if it's empty

### DISCUSSION THREAD

**Discussion by adityaladwa** [sqlc-dev/sqlc#1132] [Q&A]
### Discussed in https://github.com/kyleconroy/sqlc/discussions/1087

<div type='discussions-op-text'>

<sup>Originally posted by **adityaladwa** July 14, 2021</sup>
Let's say I have a table

```sql
create table "todos" (
    "id" uuid primary key default uuid_generate_v4(),
    "title" varchar(255) not null,
    "description" text default null
);
```

After I run `sqlc generate` the following model will be generated:

```go
type Todo struct {
	ID          uuid.UUID `json:"id"`
	Title       string    `json:"title"`
	Description sql.NullString    `json:"description"`
}
```

I would like to omit the _description_ if it's empty (as it's a nullable column). 

Now, I can use column override and use `string` type instead of `sql.NullString`. So now the generated type will be

```go
type Todo struct {
	ID          uuid.UUID `json:"id"`
	Title       string    `json:"title"`
	Description string   `json:"description"`
}
```

The problem with this is that if _Description_ is `nil`, it won't be omitted from the `json` response. So the response would be something like this:

```json
{
    "id": "c0507d40-de51-451f-b84e-c2f06af7710f",
    "title": "This is a todo",
    "description": null
}

```

Is there a way I can tell `sqlc` to add `omitempty` tag to a particular field? (Description field in this case)

Any other way to solve this issue?</div>

**Comment by efairon** [comment]
@adityaladwa do you have a solution now?

**Comment by abdullah2993** [comment]
I have opened a [PR](https://github.com/sqlc-dev/sqlc/pull/3117)  for adding support for `omitempty`
